app/api/actions/github/route.ts

```diff
@@ -467,6 +471,22 @@ export async function POST(request: Request) {
         repoName: repo,
       });
 
+      // Make Watermelon Review the PR's business logic here by comparing the title with the AI-generated summary
+      const prRating = await ratePullRequest({
```

Comment:
Do we need to do it sync?

Reply:
Async? (it's awaited)

Reply:
Precisely, we wait for it to end before running the rest. 
Can we do it in parallel? In another serverless function would be better, offload it.

Reply:
Okay, that's a good catch (making sure the PR is rated before labeling it), but I have found myself blocked trying to refactor flagPullRequest.ts to run completely sync (since it's a function that requires a lot of API calls). I updated the PR in an attempt to do so, but would need to revisit it. 

Consider the async approach though. It just works.

Reply:
"it just works". Respond please: "I am responsible for any race conditions and will fix it perpetually if network errors come up" for me to approve.

Reply:
I updated the logic to make everything work sync